frontend/src/components/Categories.tsx

```diff
@@ -1,12 +1,18 @@
 import CategoryItem from "./CategoryItem";
 import './Categories.scss'
+import {Category} from "../models/Category";
+import {ITokenConfig} from "../models/Connection";
 
-export default function Categories() {
-
+export default function Categories(props: {categories: Category[], config: ITokenConfig}) {
+    const {categories, config} = props;
+    if (!categories || !Array.isArray(categories)) return null;
```

Comment:
Since categories if from typer Categorie[], it should always be an array and never be undefined.